File: trunk/emsi/asp-net-app/WebEMSOF/component/biz/Class_biz_fy_calendar.pas
unit Class_biz_fy_calendar;

interface

uses
  Class_db_fy_calendar,
  Class_biz_milestones;

type
  TClass_biz_fy_calendar = class
  strict private
    db_fy_calendar: TClass_db_fy_calendar;
  public
    constructor Create;
  published
    function MilestoneDate(milestone: milestone_type): datetime;
  end;

implementation

constructor TClass_biz_fy_calendar.Create;
begin
  inherited Create;
  db_fy_calendar := TClass_db_fy_calendar.Create;
end;

function TClass_biz_fy_calendar.MilestoneDate(milestone: milestone_type): datetime;
begin
  MilestoneDate := db_fy_calendar.MilestoneDate(uint32(ord(milestone)).tostring);
end;

end.

File: asp-net-app/WebEMSOF/component/biz/Class_biz_fy_calendar.cs
using System;
using Class_db_fy_calendar;
using Class_biz_milestones;
namespace Class_biz_fy_calendar
{
    public class TClass_biz_fy_calendar
    {
        private TClass_db_fy_calendar db_fy_calendar = null;
        //Constructor  Create()
        public TClass_biz_fy_calendar() : base()
        {
            db_fy_calendar = new TClass_db_fy_calendar();
        }
        public DateTime MilestoneDate(milestone_type milestone)
        {
            DateTime result;
            result = db_fy_calendar.MilestoneDate(((uint)(milestone)).ToString());
            return result;
        }

    } // end TClass_biz_fy_calendar

}
